pandas/core/groupby.py

```diff
@@ -1353,6 +1353,26 @@ def var(self, ddof=1, *args, **kwargs):
             f = lambda x: x.var(ddof=ddof, **kwargs)
             return self._python_agg_general(f)
 
+    @Substitution(name='groupby')
+    @Appender(_doc_template)
+    def mad(self, skipna=True):
+        if not skipna:
+            raise NotImplementedError("'skipna=False' not yet implemented")
+
+        if self.axis != 0:
+            return self.apply(lambda x: x.mad(axis=self.axis))
+
+        # Wrap in a try..except to catch a TypeError with bool data
+        # Ideally this would be implemented in `mean` instead of here
+        try:
```

Comment:
As touched on in the comments I ideally would not want this `try...except` and think instead that the `mean` application should be throwing the error. While `mean` does raise for `object` types, something like `pd.Series([True, False, True]).mean()` is entirely valid and therefore this ends up throwing a TypeError on the subsequent `np.abs` operation.

I think `mean` should raise on boolean data and have that pass through. Will open separate issue if you agree